liquibase-extension-testing/src/main/java/liquibase/extension/testing/testsystem/TestSystemFactory.java

```diff
@@ -0,0 +1,85 @@
+package liquibase.extension.testing.testsystem;
+
+import liquibase.Scope;
+import liquibase.configuration.ConfigurationValueConverter;
+import liquibase.configuration.LiquibaseConfiguration;
+import liquibase.exception.UnexpectedLiquibaseException;
+import liquibase.plugin.AbstractPluginFactory;
+import liquibase.util.CollectionUtil;
+import liquibase.util.StringUtil;
+
+import java.lang.reflect.Constructor;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+/**
+ * Factory for getting {@link TestSystem} implementations.
+ */
+public class TestSystemFactory extends AbstractPluginFactory<TestSystem> {
+
+    private Map<TestSystem.Definition, TestSystem> systems = new HashMap<>();
```

Comment:
Should this be static? Or are we guaranteed that there will only ever be one instance of this factory?

Reply:
There should just be one instance of the factory. And if there are multiple we might want them handling their own managed systems.